repos/spack_repo/builtin/packages/mgard/package.py

```diff
@@ -26,54 +28,91 @@ class Mgard(CMakePackage, CudaPackage):
+
+    # Historical versions using fork (with patches for Spack compatibility)
+    version(
+        "compat-2023-12-09",
+        commit="d61d8c06c49a72b2e582cc02de88b7b27e1275d2",
+        git="https://github.com/robertu94/MGARD.git",
+        preferred=True,
+    )
+    version(
+        "compat-2023-03-31",
+        commit="a8a04a86ff30f91d0b430a7c52960a12fa119589",
+        git="https://github.com/robertu94/MGARD.git",
+    )
+    version(
+        "compat-2023-01-10",
+        commit="3808bd8889a0f8e6647fc0251a3189bc4dfc920f",
+        git="https://github.com/robertu94/MGARD.git",
+    )
+    version(
+        "compat-2022-11-18",
+        commit="72dd230ed1af88f62ed3c0f662e2387a6e587748",
+        git="https://github.com/robertu94/MGARD.git",
+    )
+    version(
+        "compat-2021-11-12",
+        commit="3c05c80a45a51bb6cc5fb5fffe7b1b16787d3366",
+        git="https://github.com/robertu94/MGARD.git",
+    )
+    version(
+        "compat-2020-10-01",
+        commit="b67a0ac963587f190e106cc3c0b30773a9455f7a",
+        git="https://github.com/robertu94/MGARD.git",
+    )
 
     variant(
         "serial",
-        when="@2022-11-18:",
+        when="@compat-2022-11-18:",
         default=True,
         description="Enable the classic non-parallel implmementation",
     )
-    variant("openmp", when="@2022-11-18:", default=True, description="Enable OpenMP support")
-    variant("timing", when="@2022-11-18:", default=False, description="Enable profile timings")
+    variant(
+        "openmp", when="@compat-2022-11-18:", default=True, description="Enable OpenMP support"
+    )
+    variant(
+        "timing", when="@compat-2022-11-18:", default=False, description="Enable profile timings"
+    )
     variant(
         "unstructured",
-        when="@2022-11-18:",
+        when="@compat-2022-11-18:",
         default=False,
         description="Enable experimental unstructured mesh support",
     )
 
     depends_on("c", type="build")
     depends_on("cxx", type="build")
 
-    depends_on("python", type=("build",), when="@2022-11-18:")
-    depends_on("sed", type=("build",), when="@2022-11-18:")
+    depends_on("python", type=("build",), when="@compat-2022-11-18:")
+    depends_on("sed", type=("build",), when="@compat-2022-11-18:")
     depends_on("zlib-api")
     depends_on("zlib@1.2.9:", when="^[virtuals=zlib-api] zlib")  # crc32_z
-    depends_on("pkgconfig", type=("build",), when="@2022-11-18:")
+    depends_on("pkgconfig", type=("build",), when="@compat-2022-11-18:")
     depends_on("zstd")
-    depends_on("protobuf@3.4:", when="@2022-11-18:")
+    depends_on("protobuf@3.4:", when="@compat-2022-11-18:")
     # See https://github.com/CODARcode/MGARD/issues/240
-    depends_on("protobuf@:3.28", when="@:2023-12-09")
-    depends_on("libarchive", when="@2021-11-12:")
-    depends_on("tclap", when="@2021-11-12")
-    depends_on("yaml-cpp", when="@2021-11-12:")
+    depends_on("protobuf@:3.28", when="@:compat-2023-12-09")
```

Comment:
The 1.5.2 tag happened on a commit for September 28, 2023. Thus, I expect anything that applies to version `compat-2023-12-09` should also apply to `1.5.2`.

```suggestion
    depends_on("protobuf@:3.28", when="@:1.5.2")
```

Reply:
It interestingly builds without this.

Reply:
¯\(°_o)/¯  OK. Fine by me.

Reply:
oh it appears that my local build was re-using the exsiting protobuf 3.28 from previous installation of mgard